src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -276,6 +276,20 @@ void QgsRelationReferenceWidget::setForeignKey( const QVariant& value )
   }
   else
   {
+    QVariant nullValue = QSettings().value( "qgis/nullValue", "NULL" );
```

Comment:
Is there a reason to convert this to a QVariant here and back to a QString 5 lines down?